packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -623,16 +608,18 @@ Future<void> diagnoseXcodeBuildFailure(XcodeBuildResult result, Usage flutterUsa
     logger.printError('  open ios/Runner.xcworkspace');
     return;
   }
-  if (result.stdout?.contains('Code Sign error') == true) {
-    logger.printError('');
-    logger.printError('It appears that there was a problem signing your application prior to installation on the device.');
-    logger.printError('');
-    logger.printError('Verify that the Bundle Identifier in your project is your signing id in Xcode');
-    logger.printError('  open ios/Runner.xcworkspace');
-    logger.printError('');
-    logger.printError("Also try selecting 'Product > Build' to fix the problem:");
+
+  // Handle xcresult errors.
+  final XCResult? xcResult = result.xcResult;
+  if (xcResult == null) {
     return;
   }
+  if (!xcResult.parseSuccess) {
+    globals.printTrace('XCResult parsing error: ${xcResult.parsingErrorMessage}');
+    return;
+  }
+
+  xcResult.issues.forEach( (final XCResultIssue issue) => _handleXCResultIssue(issue: issue, logger: logger));
```

Comment:
Analyzer didn't like this: 
```
info • Avoid using `forEach` with a function literal • packages/flutter_tools/lib/src/ios/mac.dart:622:19 • avoid_function_literals_in_foreach_calls
```
```suggestion
  for (final XCResultIssue issue in xcResult.issues) {
    _handleXCResultIssue(issue: issue, logger: logger);
  }
```